filebeat/prospector/prospector_log.go

```diff
@@ -56,6 +56,21 @@ func (p *ProspectorLog) Run() {
 		p.Prospector.states.Cleanup()
 		logp.Debug("prospector", "Prospector states cleaned up.")
 	}
+
+	// Cleanup of removed files will only happen after next scan. Otherwise it can happen that not all states
+	// were updated before cleanup is called
+	if p.config.CleanRemoved {
+		for _, state := range p.Prospector.states.GetStates() {
+			_, err := os.Stat(state.Source)
+			if err != nil {
+				state.TTL = 0
```

Comment:
Note for me: Lets distinguish between these two:

```
var (
        ErrPermission = errors.New("permission denied")
        ErrNotExist   = errors.New("file does not exist")
)
```

If it does not exist, I think the behaviour is clear. I suggest if we don't have access, we log an error and close the file, but we don't remove it from the registry. @urso WDYT?